services/QuillLMS/app/workers/batch_assign_recommendations_worker.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class BatchAssignRecommendationsWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: SidekiqQueue::CRITICAL
+
+  def perform(assigning_all_recommendations, pack_sequence_id, selections_with_students)
+    return if selections_with_students.empty?
+
+    last_recommendation_index = selections_with_students.length - 1
+
+    batch = Sidekiq::Batch.new
+    batch.on(:success, self.class, pack_sequence_id: pack_sequence_id)
```

Comment:
Just noting that since we are using `on(:success)`, a failed `AssignRecommendationsWorker` job will prevent any sequences from being assigned until they are successful. That's probably fine, just wanted to note it in case that was a concern.

https://github.com/mperham/sidekiq/wiki/Batches#callbacks

![Screenshot 2022-12-07 at 4 00 07 PM](https://user-images.githubusercontent.com/1304933/206294579-372ba6fb-1ed8-4e89-8c60-142625ee1edc.png)